smart-workflow-test/src_test/com/axonivy/utils/smart/workflow/model/openai/OpenAiModelIT.java

```diff
@@ -0,0 +1,47 @@
+package com.axonivy.utils.smart.workflow.model.openai;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.axonivy.utils.smart.workflow.model.ChatModelFactory.AiConf;
+import com.axonivy.utils.smart.workflow.model.openai.internal.OpenAiServiceConnector.OpenAiConf;
+import com.axonivy.utils.smart.workflow.test.TestToolUserData;
+
+import ch.ivyteam.ivy.bpm.engine.client.BpmClient;
+import ch.ivyteam.ivy.bpm.engine.client.element.BpmProcess;
+import ch.ivyteam.ivy.bpm.exec.client.IvyProcessTest;
+import ch.ivyteam.ivy.environment.AppFixture;
+import ch.ivyteam.ivy.environment.Ivy;
+import ch.ivyteam.test.log.LoggerAccess;
+import dev.langchain4j.http.client.log.LoggingHttpClient;
+
+@IvyProcessTest
+class OpenAiModelIT {
+
+  private static final BpmProcess AGENT_TOOLS = BpmProcess.name("TestToolUser");
+
+  @RegisterExtension
+  LoggerAccess log = new LoggerAccess(LoggingHttpClient.class.getName());
+
+  @BeforeEach
+  void setup(AppFixture fixture) {
+    fixture.var(AiConf.DEFAULT_PROVIDER, OpenAiModelProvider.NAME); // enforce OpenAI!
+    fixture.var(OpenAiConf.API_KEY, System.getenv("OPEN_AI_API_KEY"));
+  }
+
+  @Test
+  void structuredOutput_e2e(BpmClient client) {
+    Ivy.session().loginSessionUser("James", "secret");
+    var res = client.start()
+        .process(AGENT_TOOLS.elementName("structuredOutput"))
+        .as().session(Ivy.session())
+        .execute();
+    TestToolUserData data = res.data().last();
+    assertThat(data.getPerson().getFirstName())
+        .isEqualTo("James");
```

Comment:
a test which has it's complexity, calling tools, but still should provide very deterministic results, as we enforce a structured output of person and produce the tool-value on our terms.

... I guess the same test can be run against every other model-provider with the same results :crossed_fingers: